src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -71,21 +171,99 @@ protected override void ExecuteCore()
+                        DotNetAppHostExecutableNameWithoutExtension + ExecutableExtension.ForRuntimeIdentifier(bestAppHostRuntimeIdentifier));
+
+
+                    TaskItem appHostItem = new TaskItem("AppHost");
+                    string appHostPackPath = null;
+                    if (!string.IsNullOrEmpty(TargetingPackRoot))
+                    {
+                        appHostPackPath = GetPackPath(appHostPackName, appHostPackVersion);
+                    }
+                    if (appHostPackPath != null && Directory.Exists(appHostPackPath))
+                    {
+                        //  Use AppHost from packs folder
+                        appHostItem.SetMetadata(MetadataKeys.Path, Path.Combine(appHostPackPath, appHostRelativePathInPackage));
+                    }
+                    else
+                    {
+                        //  Download apphost pack
+                        TaskItem packageToDownload = new TaskItem(appHostPackName);
+                        packageToDownload.SetMetadata(MetadataKeys.Version, appHostPackVersion);
+                        packagesToDownload.Add(packageToDownload);
+
+                        appHostItem.SetMetadata(MetadataKeys.PackageName, appHostPackName);
+                        appHostItem.SetMetadata(MetadataKeys.PackageVersion, appHostPackVersion);
+                        appHostItem.SetMetadata(MetadataKeys.RelativePath, appHostRelativePathInPackage);
+                    }
+
+                    AppHost = new ITaskItem[] { appHostItem };
+                }
+            }
+
+            if (packagesToDownload.Any())
+            {
+                PackagesToDownload = packagesToDownload.ToArray();
+            }
+
+            if (packagesToReference.Any())
+            {
+                PackagesToReference = packagesToReference.ToArray();
             }
 
             if (runtimeFrameworks.Any())
             {
                 RuntimeFrameworks = runtimeFrameworks.ToArray();
             }
 
+            if (targetingPacks.Any())
+            {
+                TargetingPacks = targetingPacks.ToArray();
+            }
+
             if (unresolvedFrameworkReferences.Any())
             {
                 UnresolvedFrameworkReferences = unresolvedFrameworkReferences.ToArray();
             }
+        }
+
+        string GetBestRuntimeIdentifier(string targetRuntimeIdentifier, string availableRuntimeIdentifiers, out bool wasInGraph)
+        {
+            var runtimeGraph = new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath);
+            var bestRuntimeIdentifier = NuGetUtils.GetBestMatchingRid(runtimeGraph,
+                AppHostRuntimeIdentifier,
+                availableRuntimeIdentifiers.Split(';'),
+                out wasInGraph);
 
+            return bestRuntimeIdentifier;
+            
+        }
+
+        string GetPackPath(string name, string version)
```

Comment:
This should be marked static.

Also, nit: private again.

Reply:
It uses the `TargetingPackRoot` instance property, so it can't be static.